src/client/app/shared/services/name-list.service.ts

```diff
@@ -1,12 +1,37 @@
+import {Http, Response} from '@angular/http';
+import 'rxjs/add/operator/map';
+import {Injectable} from '@angular/core';
+
+@Injectable()
 export class NameListService {
-  names = [
-    'Edsger Dijkstra',
-    'Donald Knuth',
-    'Alan Turing',
-    'Grace Hopper'
-  ];
+  names:string[] = [];
+  private loading: Boolean = false;
+  constructor(private http: Http) {
+  }
+
+  requestNames() {
+      this.http.get('./assets/json/name-list.json')
+      .map(this.mapData)
+      .subscribe(this.updateNames, this.handleError);
```

Comment:
To be honest i am not sure on that, i assumed scoping issues where handled by L51, but I am not certain on this.